src/components/SVGViewer/SVGViewerSearch.tsx

```diff
@@ -0,0 +1,351 @@
+import Icon from 'antd/lib/icon';
+import React from 'react';
+import styled from 'styled-components';
+import * as CustomIcon from './Icons';
+
+const searchClassName = 'search-result';
+const currentSearchClassName = 'current-search-result';
+
+const initialState = {
+  searchPhrase: '',
+  amountOfResults: 0,
+  currentResult: 0,
+};
+
+export interface ComponentProps {
+  visible: boolean;
+  svg: SVGSVGElement;
+  isDesktop: boolean;
+  openSearch: () => void;
+  zoomOnCurrentAsset: (currentAsset: Element | null) => void;
+  handleCancelSearch: () => void;
+  addCssClassesToMetadataContainer: ({
+    condition,
+    className,
+  }: {
+    condition: (metadataNode: Element) => boolean;
+    className: string;
+  }) => void;
+  addCssClassesToSvgText: ({
+    condition,
+    className,
+  }: {
+    condition: (metadataNode: Element) => boolean;
+    className: string;
+  }) => void;
+  onFocus: () => void;
+  onBlur: () => void;
+}
+
+interface ComponentState {
+  searchPhrase: string;
+  amountOfResults: number;
+  currentResult: number;
+}
+
+class SVGViewerSearch extends React.Component<ComponentProps, ComponentState> {
+  searchInput: React.RefObject<HTMLInputElement>;
+
+  constructor(props: ComponentProps) {
+    super(props);
+    this.state = initialState;
+    this.searchInput = React.createRef();
+  }
+
+  componentDidMount() {
+    window.addEventListener('keydown', this.handleKeyDown);
```

Comment:
In order to avoid conflicts with other components while listening to keyboard events please take a look at the implementation of the key listener in `Model3DViewer` (pull master branch).

Reply:
Could you please point me to the exact line?

Reply:
Fixed using hidden input with focus/blur